src/api/app/views/webui/user/user_profile_redesign/_involvement_and_activity.html.haml

```diff
@@ -0,0 +1,38 @@
+.card{ class: CONFIG['contribution_graph'] == :off ? 'd-block' : 'd-block d-md-none' }
+  -# Header that replaces the tabs when the contribution graph is not displayed (in small viewports or when not enabled).
+  .card-header
+    %h5 Involved Projects and Packages
+  .card-body
+    = render partial: 'webui/user/involvement', locals: { user: user,
+                                                          owned: owned,
+                                                          involved_projects: involved_projects,
+                                                          involved_packages: involved_packages }
+
+-# Displaying tabs in medium-large viewports
+- unless CONFIG['contribution_graph'] == :off
+  .card{ class: CONFIG['contribution_graph'] == :off ? 'd-none' : 'd-none d-md-block' }
+    -# Tabs
+    .bg-light
+      %ul.nav.nav-tabs.pt-2.px-3.flex-nowrap.collapsible{ 'role': 'tablist' }
+        %li.nav-item
+          = link_to('#involved-projects-and-packages', id: 'involved-projects-and-packages-tab', class: 'nav-link active text-nowrap',
+                    data: { toggle: 'tab' }, role: 'tab', aria: { controls: 'involved-projects-and-packages', selected: true }) do
+            Involved Projects/Packages
+        %li.nav-item
+          = link_to('#contributions', id: 'contributions-tab', class: 'nav-link text-nowrap',
+                    data: { toggle: 'tab' }, role: 'tab', aria: { controls: 'contributions', selected: false }) do
+            Contributions
+
+    -# Tabs content
+    .card-body
+      .tab-content
+        .tab-pane.fade.show.active#involved-projects-and-packages{ role: 'tabpanel', aria: { labelledby: 'involved-projects-and-packages-tab' } }
+          = render partial: 'webui/user/involvement', locals: { user: user,
+                                                                owned: owned,
+                                                                involved_projects: involved_projects,
+                                                                involved_packages: involved_packages }
+
+        .tab-pane.fade#contributions{ role: 'tabpanel', aria: { labelledby: 'contributions-tab' } }
+          = render partial: 'webui/user/activity', locals: { activity_hash: activity_hash,
+                                                             first_day: first_day,
+                                                             last_day: last_day }
```

Comment:
Why is the logic, what get's shown and what not, done with CSS *after* you render everything?

Reply:
A bit of explanation can be useful here.

When **Contribution Graph** is **off**, the tabs and their contents are not rendered. So that part is ok (line 11 on).

However, when **Contribution Graph** is **on**, the tabs should be displayed in large viewports. But in small ones, we should only display a common card (lines 1-10). As we discussed offline and after doing a small change, the code will stay the same in this case, using CSS to display or hide the tabs.